src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.dao;
+
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can't add movie in DB: " + movie, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
+    }
+
+    @Override
+    public Optional<Movie> get(Long id) {
+        SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+        try {
+            Session session = sessionFactory.openSession();
+            return Optional.ofNullable(session.get(Movie.class, id));
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't get id: " + id, e);
```

Comment:
```suggestion
            throw new DataProcessingException("Can't get movie id: " + id, e);
```

informative exception messages